src/Common/Fingerprinting/FingerprintFactory.cs

```diff
@@ -202,24 +203,48 @@ void AddSettingToFingerprint(IReadOnlyCollection<Glob>? patterns, string setting
                     return CreateFingerprint(entries);
                 });
 
-    private void SortAndAddInputFileHashes(List<FingerprintEntry> entries, IReadOnlyList<string> files, bool pathsAreNormalized)
+    private async Task SortAndAddInputFileHashesAsync(List<FingerprintEntry> entries, IReadOnlyList<string> files, bool pathsAreNormalized)
     {
         // Sort for consistent hash ordering
         SortedDictionary<string, byte[]?> filteredNormalizedFiles = new(StringComparer.OrdinalIgnoreCase);
+        List<Task<(string, byte[]?)>> pendingHashingTasks = new();
         foreach (string file in files)
         {
             string absoluteFilePath = pathsAreNormalized ? _pathNormalizer.Unnormalize(file) : file;
-            if (_inputHasher.ContainsPath(absoluteFilePath) && !_pluginSettings.IgnoredInputPatterns.Any(pattern => pattern.IsMatch(absoluteFilePath)))
+            if (_pluginSettings.IgnoredInputPatterns.Count > 0
+                && !_pluginSettings.IgnoredInputPatterns.Any(pattern => pattern.IsMatch(absoluteFilePath)))
             {
                 string normalizedFilePath = pathsAreNormalized ? file : _pathNormalizer.Normalize(file);
-                filteredNormalizedFiles.Add(normalizedFilePath, _inputHasher.GetHash(absoluteFilePath));
+
+                ValueTask<byte[]?> hashTask = _inputHasher.GetHashAsync(absoluteFilePath);
+
+                // If the hashing task is synchronous or already complete, add it directly.
+                // Otherwise, stash the task for later.
+                if (hashTask.IsCompleted)
+                {
+                    byte[]? hash = hashTask.Result;
+                    filteredNormalizedFiles.Add(normalizedFilePath, hash);
+                }
+                else
+                {
+                    pendingHashingTasks.Add(WrapHashingTask(normalizedFilePath, hashTask.AsTask()));
+                }
             }
         }
 
```

Comment:
`await Task.WhenAll(...)`

if you want to not miss any exceptions

Reply:
I had intentionally done it this way to avoid the extra allocation, but I may be being too stingy with allocations in this case.